pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/PresentationServiceImpl.scala

```diff
@@ -585,6 +927,77 @@ private class PresentationServiceImpl(
       PresentationRecord.ProtocolState.PresentationVerificationFailed
     )
 
+  override def verifyAnoncredPresentation(
+      presentation: Presentation,
+      requestPresentation: RequestPresentation,
+      recordId: DidCommID
+  ): ZIO[WalletAccessContext, PresentationError, PresentationRecord] = {
+    for {
+      serializedPresentation <- presentation.attachments.head.data match {
+        case Base64(data) => ZIO.succeed(AnoncredPresentation(new String(JBase64.getUrlDecoder.decode(data))))
+        case _            => ZIO.fail(InvalidAnoncredPresentation("Expecting Base64-encoded data"))
+      }
+      deserializedPresentation <-
+        AnoncredPresentationV1.schemaSerDes
+          .deserialize(serializedPresentation.data)
+          .mapError(error => PresentationError.UnexpectedError(error.error))
+      schemaIds = deserializedPresentation.identifiers.map(_.schema_id)
+      schemaMap <-
+        ZIO
+          .collectAll(schemaIds.map { schemaId =>
+            resolveSchema(schemaId)
+          })
+          .map(_.toMap)
+      credentialDefinitionIds = deserializedPresentation.identifiers.map(_.cred_def_id)
+      credentialDefinitionMap <-
+        ZIO
+          .collectAll(credentialDefinitionIds.map { credentialDefinitionId =>
+            for {
+              guid <-
+                ZIO
+                  .fromOption(CredentialDefinition.extractGUID(credentialDefinitionId))
+                  .mapError(_ =>
+                    InvalidAnoncredPresentation(s"CredentialDefinitionId format invalid ${credentialDefinitionId}")
+                  )
+              credentialDefinition <-
+                credentialDefinitionService
+                  .getByGUID(guid)
+                  .mapError(e => UnexpectedError(e.toString))
```

Comment:
Should it be some other type of error? For instance, `CredentialDefitionNotFoundError`

Reply:
Code was changes not applicable anymore